dns/jsonify/main.go

```diff
@@ -0,0 +1,218 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/kubernetes/test/e2e/perftype"
+)
+
+const (
+	// S_TO_MS is a second to milisecond ratio.
+	S_TO_MS = 1000
+)
+
+// BenchamrkResult is a dns benchmark results structure.
+type BenchamrkResult struct {
+	Code   int                    `json: code`
+	Data   map[string]interface{} `json: data`
+	Params map[string]interface{} `json: params`
+}
+
+func main() {
+	err := run()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func run() error {
+	var benchmarkDirPath string
+	var jsonDirPath string
+	var benchmarkName string
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	flag.StringVar(&benchmarkDirPath, "benchmarkDirPath", ".", "benchmark results directory path")
+	flag.StringVar(&jsonDirPath, "jsonDirPath", ".", "json results directory path")
+	flag.StringVar(&benchmarkName, "benchmarkName", ".", "benchmark name")
+
+	if err := flag.CommandLine.Parse(os.Args[1:]); err != nil {
+		return fmt.Errorf("flag parse failed: %v", err)
+	}
+
+	fmt.Printf("benchmarkDirPath: %v\n", benchmarkDirPath)
+	fmt.Printf("jsonDirPath: %v\n", jsonDirPath)
+	fmt.Printf("benchmarkName: %v\n", benchmarkName)
+
+	latency := perftype.PerfData{Version: "v1"}
+	latencyPerc := perftype.PerfData{Version: "v1"}
+	queries := perftype.PerfData{Version: "v1"}
+
+	fileList, err := getFileList(benchmarkDirPath)
+	if err != nil {
+		return fmt.Errorf("listing files error: %v", err)
+	}
+
+	for _, file := range fileList {
+		fmt.Printf("processing %v\n", file)
+		result, err := readBenchamrkResult(filepath.Join(benchmarkDirPath, file))
+		if err != nil {
+			return err
+		}
+		latency.DataItems = appendLatency(latency.DataItems, result)
+		latencyPerc.DataItems = appendLatencyPerc(latencyPerc.DataItems, result)
+		queries.DataItems = appendQueries(queries.DataItems, result)
+	}
+
+	timeString := time.Now().Format(time.RFC3339)
+	if err = saveMetric(&latency, filepath.Join(jsonDirPath, "Latency_"+benchmarkName+"_"+timeString+".json")); err != nil {
+		return err
+	}
+	if err = saveMetric(&latencyPerc, filepath.Join(jsonDirPath, "LatencyPerc_"+benchmarkName+"_"+timeString+".json")); err != nil {
+		return err
+	}
+	if err = saveMetric(&queries, filepath.Join(jsonDirPath, "Queries_"+benchmarkName+"_"+timeString+".json")); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getFileList(dir string) ([]string, error) {
```

Comment:
// getFileList returns a list of all files with extension `.out`

Reply:
Done.